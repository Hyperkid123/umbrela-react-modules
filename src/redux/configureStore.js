import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
