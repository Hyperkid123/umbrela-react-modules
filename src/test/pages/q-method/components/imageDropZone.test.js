import React from 'react'
import ImageDropZone, {ImageDropZone as Snapshot} from '../../../../pages/q-method/components/imageDropZone';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('ImageDropZone component', () => {
  const initialState  = {
    qMethodBuilder: {
      present: {
        images: []
      }
    }
  };

  const mockStore = configureStore();
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<ImageDropZone store={store}/>)
  });

  it('should render correclty' , () => {
      const tree = shallow(
        <Snapshot/>
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });
});
