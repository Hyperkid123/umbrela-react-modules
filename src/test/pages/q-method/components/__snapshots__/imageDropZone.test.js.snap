// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDropZone component should render correclty 1`] = `
<div
  className="sc-bdVaJa hDcTMv"
>
  <t
    accept="image/*"
    activeStyle={
      Object {
        "background": "repeating-linear-gradient(
            45deg,
            #eeeeee,
            #eeeeee 10px,
            #e0e0e0 10px,
            #e0e0e0 20px
          )",
      }
    }
    disableClick={false}
    disablePreview={false}
    disabled={false}
    maxSize={Infinity}
    minSize={0}
    multiple={true}
    preventDropOnDocument={true}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#757575",
        "borderRadius": 5,
        "borderStyle": "dashed",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 200,
        "padding": 5,
        "width": "auto",
      }
    }
  >
    <h2>
      Přetáhněte soubor obrázku zde nebo klikněte.
    </h2>
    <pure(CloudUpload) />
  </t>
  <StimulPreview />
</div>
`;
